lib/Handler.js

```diff
@@ -148,17 +164,15 @@ class Handler {
    *   which resolves with it.
    */
   process() {
-    return this.processor(this.event, this.context);
+    return this.processor(this);
   }
 
   /**
    * Perform cleanup tasks before responding.
    */
   cleanup() {
-    if (this.options.enableProfiling) {
-      const profile = this.constructor.profiler.stopProfiling(this.context.awsRequestId);
-      this.profile = this.constructor.deflateSync(JSON.stringify(profile)).toString('base64');
-      profile.delete();
+    if (this.profilingEnabled) {
+      this.stopProfiling();
```

Comment:
If it were me, I'd always call `stopProfiling` and just have it be a noop when profiling is not enabled